### PR TITLE
[Debt] Remove `updatePoolCandidateAsAdmin`

### LINKED ISSUE #9073: ♻️ Break up `updatePoolCandidateAsAdmin` mutation
## ♻️ Debt/Refactor

Break up updatePoolCandidateAsAdmin into separate mutations for status (with expiry date) and for notes, since they now have separate permissions.

Toggle bookmark already has its own mutation.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the debt was addressed and expected functionality has not been affected.

- [ ] updatePoolCandidateAsAdmin mutation is removed
- [ ] changes are invisible on the frontend
- [ ] new mutations can be controlled by a single, simple policy
- [ ] remove the updateAsAdmin policy

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #9073 

## 👋 Introduction

Mutations for bookmark and notes already existed, this takes care of the last bit remaining. 

## 🕵️ Details

I left the `cmoIdentifier` in, I don't think it is used anywhere though so should I also mark it deprecated?

## 🧪 Testing

1. Can update status and expiry as normal, say, by going to a review application page and using the record final decision dialog

### CODE REVIEW

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -1609,18 +1608,18 @@ type Mutation {
...
     @update
     @guard
-    @canFind(ability: "updateAsAdmin", find: "id", injectArgs: true)
+    @canFind(ability: "updateStatus", find: "id", injectArgs: true)
   # Return a boolean to prevent automatically updating the client's cache
   togglePoolCandidateBookmark(id: ID!): Boolean
     @guard
     @canFind(
-      ability: "updateAsAdmin"
+      ability: "updateStatus"
```
Comment (tristan-orourke) [inline api/graphql/schema.graphql:1622]: Could you create a new method in the policy for updateBookmark? Even if we leverage another permission there, that seems like the right level of abstraction.

**Review by tristan-orourke — CHANGES_REQUESTED** [review]

**Comment by tristan-orourke** [comment]
> I think I don't really understand the reasoning behind the issue

@petertgiles fair, this PR involves fewer changes than I expected when I original wrote the issue. But honestly my main motivation was to be able to delete the `updateAsAdmin` policy method, which is too broad, and I'm very happy to see it replaced with more specific auth scopes here.

**Comment by Jerryescandon** [comment]
I asked both DCM and Gray about the `CMO identifier` field and got approval to deprecate. 👍 Trash it! 🗑️

File: api/app/Policies/PoolCandidatePolicy.php [hunk 2]
```diff
@@ -205,6 +180,12 @@ public function viewStatus(User $user, PoolCandidate $poolCandidate)
         return $user->isAbleTo('view-team-applicationStatus', $poolCandidate->pool->team);
     }
 
+    /**
+     * Determine whether the user can update status fields for a pool candidate
+     * Note: this refers to a pool candidate's status, expiry, and bookmarked fields together
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/app/Policies/PoolCandidatePolicy.php]: Sure can do https://github.com/GCTC-NTGC/gc-digital-talent/pull/9696/commits/03944f36cd124ba0d47b0a5bf28bf8687aaccac7

File: api/graphql/schema.graphql (hunk 1 above)
Comment (vd1992) [reply api/graphql/schema.graphql]: https://github.com/GCTC-NTGC/gc-digital-talent/pull/9696/commits/03944f36cd124ba0d47b0a5bf28bf8687aaccac7

File: api/graphql/schema.graphql [hunk 3]
```diff
@@ -1609,18 +1608,18 @@ type Mutation {
     @guard
     @canModel(ability: "create")
     @injectNow(name: "submitted_at")
-  updatePoolCandidateAsAdmin(
+  updatePoolCandidateStatusAndExpiry(
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/graphql/schema.graphql]: Sure, why not I suppose 🤷 
https://github.com/GCTC-NTGC/gc-digital-talent/pull/9696/commits/85f79417b10c63f858a6d13cbda6fa272ded82fc

**Review by tristan-orourke — APPROVED** [review]
LGTM! Happy to see that "updateAsAdmin" stuff go.

**Pull request merged**